src/coreComponents/common/format/table/TableLayout.cpp

```diff
@@ -71,9 +110,153 @@ integer const & TableLayout::getColumnMargin() const
   return m_columnMargin;
 }
 
+integer const & TableLayout::getMarginValue() const
+{
+  return m_marginValue;
+}
+
 integer const & TableLayout::getMarginTitle() const
 {
   return m_titleMargin;
 }
 
+std::vector< size_t > & TableLayout::getSublineInHeaderCounts()
+{
+  return m_sublineHeaderCounts ;
+}
+
+std::vector< size_t > & TableLayout::getNbSubDataLines()
+{
+  return m_sublineDataCounts ;
+}
+
+void divideCell( std::vector< string > & lines, string const & value )
+{
+  std::istringstream strStream( value );
+  std::string line;
+  lines.clear();
+  while( getline( strStream, line, '\n' ))
+  {
+    lines.push_back( line );
+  }
+
+  if( line.empty())
+  {
+    lines.push_back( "" );
+  }
+}
+
+TableLayout::CellLayout::CellLayout():
+  m_lines( {""} ),
+  m_cellType( CellType::Header ),
+  m_alignment( TableLayout::Alignment::center ),
+  m_maxDataLength( 0 )
+{}
+
+TableLayout::CellLayout::CellLayout( CellType type, string const & cellValue, TableLayout::Alignment m_cellAlignment ):
+  m_cellType( type ),
+  m_alignment( m_cellAlignment )
+{
+  divideCell( m_lines, cellValue );
+  if( !m_lines.empty())
+  {
+    m_maxDataLength = std::max_element( m_lines.begin(), m_lines.end(), []( const auto & a, const auto & b )
+    {
+      return a.length() < b.length();
+    } )->length();
+  }
+  else
+  {
+    m_maxDataLength = 0;
+  }
+}
+
+TableLayout::Column::Column():
+  m_parent( nullptr ), m_next( nullptr ), m_maxStringSize( 0 )
+{
+  m_columName.m_lines = {};
+  m_columName.m_cellType  = CellType::Header;
+  m_columName.m_alignment = Alignment::center;
+}
+
+TableLayout::Column::Column( TableLayout::CellLayout cell ):
+  m_parent( nullptr ), m_next( nullptr ), m_maxStringSize( cell.m_maxDataLength )
+{
+  m_columName = cell;
+}
+
+
+TableLayout::Column & TableLayout::Column::setName( string_view name )
+{
+  m_columName.m_lines.push_back( std::string( name ) );
+  divideCell( m_columName.m_lines, m_columName.m_lines[0] );
+  m_columName.m_cellType = CellType::Header;
+  return *this;
+}
+
+TableLayout::Column & TableLayout::Column::hide()
+{
+  m_columName.m_cellType = CellType::Hidden;
+  return *this;
+}
+
+void TableLayout::Column::setMaxStringSize( size_t const size )
+{
+  m_maxStringSize = size;
+}
+
+size_t TableLayout::Column::getMaxStringSize() const
+{
+  return m_maxStringSize;
+}
```

Comment:
- `m_maxDataLength` not init in `Column::Column().setName()` initialization scenario,
- this data duplication may not be useful, can lead to further complication,
- it only inform about the column instance header (not subColumns / parent), so we could have a clearer name.

To sum-up, considering my other suggested renaming:
```suggestion
size_t TableLayout::Column::getHeaderMaxLineWidth() const
{
  return m_header.m_maxLineWidth;
}
```
+ remove `m_maxStringSize`?

Reply:
removed